fetch/range/blob.any.js

```diff
@@ -0,0 +1,100 @@
+  return fetch(URL.createObjectURL(blob), {
+    "method": "GET",
+    "headers": {
+      "Range": "bytes=9-21"
+    }
+  }).then(function(resp) {
+    assert_equals(resp.status, 206, "HTTP status is 206");
+    assert_equals(resp.type, "basic", "response type is basic");
+    assert_equals(resp.headers.get("Content-Type"), "text/plain", "Content-Type is " + resp.headers.get("Content-Type"));
+    assert_equals(resp.headers.get("Content-Length"), "13", "Content-Length is " + resp.headers.get("Content-Length"));
+    assert_equals(resp.headers.get("Content-Range"), "bytes 9-21/30", "Content-Range is " + resp.headers.get("Content-Range"));
+    return resp.text();
+  }).then(function(text) {
+    assert_equals(text, 'Hello, World!', "Response's body is 'Hello, World!'");
+  });
+}, "Blob range request slices the blob");
+
+promise_test(function(test) {
+  var blob = new Blob(["Multiple ranges are not currently supported"], { "type" : "text/plain" });
+  var promise = fetch(URL.createObjectURL(blob), {
+    "method": "GET",
+    "headers": {
+      "Range": "bytes=0-5,15-"
+    }
+  });
+  return promise_rejects_js(test, TypeError, promise);
+}, "Blob range request with multiple range values");
+
+promise_test(function(test) {
+  var blob = new Blob(["Range end should be greater than range start"], { "type" : "text/plain" });
+  var promise = fetch(URL.createObjectURL(blob), {
+    "method": "GET",
+    "headers": {
+      "Range": "bytes=10-5"
+    }
+  });
+  return promise_rejects_js(test, TypeError, promise);
+}, "Blob range request with short range end");
+
+promise_test(function(test) {
+  var blob = new Blob(["Range with no start or end"], { "type" : "text/plain" });
+  var promise = fetch(URL.createObjectURL(blob), {
+    "method": "GET",
+    "headers": {
+      "Range": "bytes=-"
+    }
+  });
+  return promise_rejects_js(test, TypeError, promise);
+}, "Blob range with no start or end");
+
+promise_test(function(test) {
+  var blob = new Blob(["Range with no end"], { "type" : "text/plain" });
+  var promise = fetch(URL.createObjectURL(blob), {
+    "method": "GET",
+    "headers": {
+      "Range": "bytes=10-",
+    }
+  });
+  return promise.then(function(resp) {
+    assert_equals(resp.status, 206, "HTTP status is 206")
+    assert_equals(resp.type, "basic", "response type is basic");
+    assert_equals(resp.headers.get("Content-Type"), "text/plain", "Content-Type is " + resp.headers.get("Content-Type"));
+    assert_equals(resp.headers.get("Content-Length"), "7", "Content-Length is " + resp.headers.get("Content-Length"));
+    assert_equals(resp.headers.get("Content-Range"), "bytes 10-16/17", "Content-Range is " + resp.headers.get("Content-Range"));
+  });
+}, "Blob range with no end");
+
+promise_test(function(test) {
+  var blob = new Blob(["Range with no start"], { "type" : "text/plain" });
+  var promise = fetch(URL.createObjectURL(blob), {
+    "method": "GET",
+    "headers": {
+      "Range": "bytes=-10",
```

Comment:
This will likely kinda depend on https://github.com/whatwg/fetch/issues/1450

Reply:
Since browsers support this I guess we should make the parser support this in general.